hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -815,6 +815,12 @@ public AbstractEntityPersister(
 			}
 		}
 
```

Comment:
Is the following check sufficient? It looks like `EntityMetamodel#isInstrumented` only checks for instrumentation for lazy loading. Does there need to be some other check for other types of enhancement?